Improve return value of `BootMouse.update`

relic-se opened the pull request:
Relatively simple fix here. The changes are as follows:
- `BootMouse.update` returns a `tuple` (immutable) rather than a `list` object
- If the device reads successfully and no buttons are pressed, an empty `tuple` is returned rather than `None`
- If an error is encountered during the device read, `None` is returned

Fixes https://github.com/adafruit/Adafruit_CircuitPython_USB_Host_Mouse/issues/8

FoamyGuy reviewed (approved):
Thanks for the improvements! This looks good to me, I tested successfully on Fruit Jam 10.0.0-beta.3 with the simpletest from this repo.

If it turns out that creating tuples in the main loop ends up leading to excessive GC action, it may be worth considering changing `update()` to return an int with bits that correspond to the buttons that are pressed instead of a list. It would make the API a little less "pythonic" but would mean that there is no tuple or list getting created by it in the main loop.

relic-se commented:
> If it turns out that creating tuples in the main loop ends up leading to excessive GC action, it may be worth considering changing `update()` to return an int with bits that correspond to the buttons that are pressed instead of a list. It would make the API a little less "pythonic" but would mean that there is no tuple or list getting created by it in the main loop.

My _(potentially incorrect)_ understanding was that tuples were more performant and used less memory than lists, but yes, using an integer would be even better.

How I managed this in my recent `relic_usb_host_gamepad` library was to keep a private integer within the object which is updated during `update`. Then, I added boolean properties which test the integer with a bitmask to determine if the button was pressed when the mouse was last updated. _Since gamepads have a whole lot of buttons, I got a little more fancy using descriptors and changed states, but that might not be necessary here._

This would look something along the lines of:
``` python
while True:
    if mouse.update():  # True if successfully read from mouse
        if mouse.left_button:  # tests against internal integer
            print("left button pressed")
        if mouse.right_button:
            print("right button pressed")
```

If this is how you'd like to see this working, I could draft up something along these lines.
